### PR TITLE
Add Gen 7 Battle Factory

### PR DESCRIPTION
**Pull request by LegoFigure11**

### CODE REVIEW

File: data/random-teams.js [hunk 1]
```diff
@@ -2537,6 +2539,247 @@ class RandomTeams extends Dex.ModdedDex {
...
+		let teamData = {typeCount: {}, typeComboCount: {}, baseFormes: {}, megaCount: 0, zCount: 0, has: {}, forceResult: forceResult, weaknesses: {}, resistances: {}};
+		let requiredMoveFamilies = ['hazardSet', 'hazardClear'];
+		let requiredMoves = {'stealthrock': 'hazardSet', 'rapidspin': 'hazardClear', 'defog': 'hazardClear'};
+		let weatherAbilitiesSet = {'drizzle': 'raindance', 'drought': 'sunnyday', 'snowwarning': 'hail', 'sandstream': 'sandstorm'};
+		let resistanceAbilities = {
+			'dryskin': ['Water'], 'waterabsorb': ['Water'], 'stormdrain': ['Water'],
+			'flashfire': ['Fire'], 'heatproof': ['Fire'],
+			'lightningrod': ['Electric'], 'motordrive': ['Electric'], 'voltabsorb': ['Electric'],
+			'sapsipper': ['Grass'],
+			'thickfat': ['Ice', 'Fire'],
```
Comment (AustinXII) [inline data/random-teams.js]: Is this one intended? The other abilities actually make it so the move won’t do any damage while this one is actually a resistance abilities

**Comment by Zarel** [comment]
I thought this was already added? @TheImmortal?

**Comment by KrisXV** [comment]
@zarel the only gen 7 battle factory format is BSS factory currently, the other battle factory that’s currently implemented is gen 6

**Comment by Zarel** [comment]
Oh, okay. I'll let TI decide what to do with this.

File: data/random-teams.js (hunk 1 above)
Comment (LegoFigure11) [reply data/random-teams.js]: It was listed alongside these other abilities in the original Gen 6 Battle Factory, so I figured it should probably be here too.

**Comment by Zarel** [comment]
so people have reported two bugs so far:

- Houndoom has "houndoomite" instead of "Houndoominite"
- Ninetales-Aurora has the ability "auroraveil", which is actually a move

**Comment by Zarel** [comment]
Should we write a linter for Battle Factory sets?

**Comment by KrisXV** [comment]
crawdaunt has ada**tp**ability instead of ada**pt**ability

**Comment by Zarel** [comment]
Also why do factory sets have move, ability, and item names in arrays?

**Comment by KrisXV** [comment]
They're in arrays for the possibility of slashes on the set, like how Sylveon can run multiple hidden powers in monotype on its Choice Specs set

**Pull request merged**